commit 4fefcf517133260a7b63049d3a02c9249fe7776c
Author: Ingo Franzki <ifranzki@linux.ibm.com>
Date:   Mon Apr 15 09:31:12 2024 +0200

    EP11: Fix compile error with NO_PKEY defined
    
    Function signature of ep11tok_pkey_usage_ok() has changed, also change the
    code inside the #ifdef NO_PKEY block.
    
    Fixes: cf978b111205b206c7b3c53f424f7085913c00d0
    
    Signed-off-by: Ingo Franzki <ifranzki@linux.ibm.com>

diff --git a/usr/lib/ep11_stdll/ep11_specific.c b/usr/lib/ep11_stdll/ep11_specific.c
index b5d788bf..e9007a16 100644
--- a/usr/lib/ep11_stdll/ep11_specific.c
+++ b/usr/lib/ep11_stdll/ep11_specific.c
@@ -1460,15 +1460,15 @@ done:
 }
 
 #ifdef NO_PKEY
-CK_BBOOL ep11tok_pkey_usage_ok(STDLL_TokData_t *tokdata, SESSION *session,
-                               CK_OBJECT_HANDLE hkey, CK_MECHANISM *mech)
+CK_RV ep11tok_pkey_usage_ok(STDLL_TokData_t *tokdata, SESSION *session,
+                            CK_OBJECT_HANDLE hkey, CK_MECHANISM *mech)
 {
     UNUSED(tokdata);
     UNUSED(session);
     UNUSED(hkey);
     UNUSED(mech);
 
-    return CK_FALSE;
+    return CKR_FUNCTION_NOT_SUPPORTED;
 }
 #endif /* NO_PKEY */
 
